src/redux/downloadevent/downloadEventMiddleware.tsx

```diff
@@ -0,0 +1,198 @@
+                message={i18n.t('download-event.message.failed', {
+                  ns: 'notifications',
+                  file: data.id
+                })}
+                downloadProgress={50}
+                color={color}
+                animated={false}
+              />
+            ),
+            icon: <ErrorIcon size={IconSize.sm} />,
+            autoClose: true,
+            loading: false,
+            color: color
+          });
+
+          delete downloads[data.id];
+        }
+
+        if (data.type === DownloadType.Started && !downloads[data.id]) {
+          notifications.show({
+            id: data.id,
+            title: i18n.t('download-event.title.started', {
+              ns: 'notifications',
+              file: data.id
+            }),
+            message: (
+              <DownloadEventNotificationBody
+                message={i18n.t('download-event.message.started', {
+                  ns: 'notifications',
+                  file: data.id
+                })}
+                downloadProgress={0}
+              />
+            ),
+            autoClose: false,
+            loading: true,
+            color: 'white'
+          });
+          downloads[data.id] = data.id;
+        }
+
+        if (data.type === DownloadType.Progress) {
+          if (downloads[data.id]) {
+            notifications.update({
+              id: data.id,
+              title: i18n.t('download-event.title.progress', {
+                ns: 'notifications',
+                file: data.id
+              }),
+              message: (
+                <DownloadEventNotificationBody
+                  message={i18n.t('download-event.message.progress', {
+                    ns: 'notifications',
+                    file: data.id
+                  })}
+                  downloadedSize={downloadedMb}
+                  totalSize={totalMb}
+                />
+              ),
+              autoClose: false,
+              loading: true,
+              color: 'white'
+            });
+          } else {
+            // We've already started downloading but frontend may not have been connected
+            // yet, or Gui was refreshed during download
+            notifications.show({
+              id: data.id,
+              title: i18n.t('download-event.title.progress', {
+                ns: 'notifications',
+                file: data.id
+              }),
+              message: (
+                <DownloadEventNotificationBody
+                  message={i18n.t('download-event.message.progress', {
+                    ns: 'notifications',
+                    file: data.id
+                  })}
+                  downloadedSize={downloadedMb}
+                  totalSize={totalMb}
+                />
+              ),
+              autoClose: false,
+              loading: true,
+              color: 'white'
+            });
+            downloads[data.id] = data.id;
+          }
+        }
+      }
```

Comment:
Should these really be a series of `if` statements? As far as i understand it, shouldn't only one of these if statements run for each time? Doesn't it make more sense with a switch, or `else if`s for type and then in the `DownloadType.Started` have an if statement with an exit for the `!downloads[data.id]` line?